src/saltext/vmware/modules/dvswitch.py

```diff
@@ -281,3 +282,155 @@ def manage(
         vmodl.MethodFault,
     ) as exc:
         raise salt.exceptions.SaltException(str(exc))
+
+
+def remove_host(
+    switch_name,
+    host_name=None,
+    datacenter_name=None,
+    cluster_name=None,
+    service_instance=None,
+):
+    """
+    Remove an ESXi host from a distributed vSwitch.
+
+    switch_name
+        Name of the distributed vSwitch.
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    service_instance
+        Use this vCenter service connection instance instead of creating a new one. (optional).
+
+    .. code-block:: bash
+
+        salt '*' vmware_dvswitch.remove_host switch_name=dvs1 hostname=host1
+
+    """
+    log.debug("Running vmware_dvswitch.remove_host")
+    ret = {}
+    if not service_instance:
+        service_instance = get_service_instance(opts=__opts__, pillar=__pillar__)
+    hosts = utils_esxi.get_hosts(
+        service_instance=service_instance,
+        host_names=[host_name] if host_name else None,
+        cluster_name=cluster_name,
+        datacenter_name=datacenter_name,
+        get_all_hosts=True if not host_name else False,
+    )
+    try:
+        for h in hosts:
+            ret[h.name] = False
+            if hasattr(h.configManager.networkSystem.networkInfo, "vswitch"):
+                h.configManager.networkSystem.RemoveVirtualSwitch(vswitchName=switch_name)
+                ret[h.name] = True
+        return ret
+    except (
+        vim.fault.InvalidState,
+        vim.fault.NotFound,
+        vim.fault.HostConfigFault,
+        vim.fault.ResourceInUse,
+        vmodl.fault.InvalidArgument,
+        salt.exceptions.VMwareApiError,
+    ) as exc:
+        raise salt.exceptions.SaltException(str(exc))
+
+
+def manage_host(
+    switch_name,
+    host_name=None,
+    datacenter_name=None,
+    cluster_name=None,
+    nics=None,
+    num_ports=None,
+    mtu=None,
+    service_instance=None,
+):
+    """
+    Add/update an ESXi host to a distributed vSwitch.
+
+    switch_name
+        Name of the distributed vSwitch.
+
+    datacenter_name
+        Filter by this datacenter name (required when cluster is specified)
+
+    cluster_name
+        Filter by this cluster name (optional)
+
+    host_name
+        Filter by this ESXi hostname (optional)
+
+    nics
+        List of vmnics to attach to vSwtich. (optional). Default "None".
+
+    num_ports
+        Number of ports to configure on the vSwitch. (optional). Default 128.
+
+    mtu
+        MTU to configure on the vSwitch. (optional). Default 1600.
```

Comment:
we can reconfigure the MTU? I suspect we're configuring a virtual NIC's MTU, not the vSwitch's MTU?

Reply:
same as above